integration/test/Test/Bot.hs

```diff
@@ -0,0 +1,142 @@
+{-# OPTIONS_GHC -Wwarn #-}
+
+module Test.Bot where
+
+import API.Brig
+import API.Common
+import API.Galley
+import Control.Lens hiding ((.=))
+import qualified Data.Aeson as Aeson
+import qualified Data.ProtoLens as Proto
+import Data.String.Conversions (cs)
+import Network.HTTP.Types (status200, status201)
+import Network.Wai (responseLBS)
+import qualified Network.Wai as Wai
+import qualified Network.Wai.Route as Wai
+import Numeric.Lens (hex)
+import qualified Proto.Otr as Proto
+import qualified Proto.Otr_Fields as Proto
+import SetupHelpers
+import Testlib.Certs
+import Testlib.MockIntegrationService
+import Testlib.Prelude
+import UnliftIO
+
+{- FIXME(mangoiv):
+ -
+ - The issue is as follows:
+ -
+ - certificate validation should work only for self-signed certs, this is checked by the signature
+ - verification function; so this test fails if there's any unknown entity (CA) involved who
+ - signed the cert. (a cert can only have one signatory, a CA or self)
+ -
+ - this test succeeds if the signature verification fails (because it's not self signed), however,
+ - even if Brig starts to do signature verification, the test would still succeed, because brig
+ - doesn't know (or trust) the CA, anyway, even if it does signature verification.
+ -
+ - For this test to make sense, we would have to make sure that the brig we're testing against
+ - *would* trust the CA, *if* it did verification, because only in that case it would now succeed
+ - with verification and not return a "PinInvalidCert" error.
+ - -}
```

Comment:
due to this, I'm a bit stuck on this PR, most importantly I don't know how I would distinguish between failing due to not knowing the CA and failing due to not doing CA verification without modifying Brig. 

This additionally implies, that I do know how to distinguish the cases of 
- root CA
- intermediate CA
- leaf cert (which is conceptually the same as above)

Reply:
> For this test to make sense, we would have to make sure that the brig we're testing against
 *would* trust the CA, *if* it did verification, because only in that case it would now succeed
 with verification and not return a "PinInvalidCert" error.

If brig starts doing some CA verification tomorrow, the people will hopefully write a test to make sure that that verification works. This test failing would just start meaning that the unknown CAs are not trusted. So, IMO this test is good as it is.

Reply:
Maybe we can call this test `testBotUnknownRootCA` anyway.

Reply:
This test would start failing if the CAs were made trusted *and* if the signature verification would be done, at the same time, so if we implemented this and then went back to modify the test such that brig starts trusting these CAs, then it would start failing.